cppcache/include/geode/Cache.hpp

```diff
@@ -222,14 +222,25 @@ class APACHE_GEODE_EXPORT Cache : public GeodeCache {
   /**
    * Returns a factory that can create a {@link PdxInstance}.
    * @param className the fully qualified class name that the PdxInstance will
-   * become
-   * when it is fully deserialized.
+   * become when it is fully deserialized.
    * @throws IllegalStateException if the className is nullptr or invalid.
    * @return the factory
    */
   PdxInstanceFactory createPdxInstanceFactory(
       const std::string& className) const override;
 
+  /**
+   * Returns a factory that can create a {@link PdxInstance}.
+   * @param className the fully qualified class name that the PdxInstance will
+   * become when it is fully deserialized.
+   * @param expectDomainClass Whether or not created PdxType represents a
+   * Java domain class.
+   * @throws IllegalStateException if the className is nullptr or invalid.
+   * @return the factory
+   */
+  PdxInstanceFactory createPdxInstanceFactory(
+      const std::string& className, bool expectDomainClass) const override;
```

Comment:
Is the domain class thing actually specific to Java? I could also apply to .NET or other languages where a reflection like behavior exists to instantiate a type by name dynamically. If it is specific to Java then maybe the param name should reflect that. If it isn't specific to Java then maybe the documentation should change.

Reply:
Short answer -> Currently, it's specific to Java, so yes, will update the variable name. Thanks for pointing it out
And it's not directly involved with reflection. You can have custom types deserialization and not have reflection. You just need to tell your application how to serialize/de-serialize objects.